File: src/main/java/net/starlegacy/feature/gas/collectionfactors/WorldLimitFactor.java
package net.starlegacy.feature.gas.collectionfactors;

import java.util.Arrays;
import org.bukkit.Location;

public class WorldLimitFactor extends CollectionFactor {

    private String[] enabledWorlds;

    public WorldLimitFactor(String[] enabledWorlds) {
        this.enabledWorlds = enabledWorlds;
    }

    @Override
    public boolean factor(Location location) {
        return Arrays.stream(enabledWorlds).anyMatch(location.getWorld().getName()::equalsIgnoreCase);
    }
}

File: server/src/main/kotlin/net/horizonsend/ion/server/features/gas/collectionfactors/WorldLimitFactor.kt
package net.horizonsend.ion.server.features.gas.collectionfactors

import org.bukkit.Location

class WorldLimitFactor(private val enabledWorlds: List<String>) : CollectionFactor() {
	override fun factor(location: Location): Boolean {
		return enabledWorlds.any { allowedWorld -> location.world.name.equals(allowedWorld, ignoreCase = true) }
	}

	override fun canBeFound(location: Location): Boolean = factor(location)
}
